lib/constable/models/announcement.ex

```diff
@@ -39,32 +39,41 @@ defmodule Constable.Announcement do
   end
 
   def with_announcement_list_assocs(query \\ __MODULE__) do
-    from q in query, preload: [
-      :user,
-      interests: ^Interest.ordered_by_name,
-      comments: ^newest_comments_first(),
-      comments: :user
-    ]
+    from q in query,
+      preload: [
+        :user,
+        interests: ^Interest.ordered_by_name(),
+        comments: ^newest_comments_first(),
+        comments: :user
+      ]
+  end
+
+  def for_user(query \\ __MODULE__, user_id) do
+    query
+    |> where([a], a.user_id == ^user_id)
   end
 
   def search(query \\ __MODULE__, search_term, exclude_interests: excludes) do
     search_term = search_term |> prepare_for_tsquery
 
-    excludes = if Enum.empty?(excludes) do
-      [""]
-    else
-      excludes
-    end
+    excludes =
```

Comment:
Nitpick: What would you think about adding a function head to remove this conditional?

```elixir
def search(query \\ __MODULE__, search_term, exclude_interests: []) do
  search(query, search_term, exclude_interests: [""]
end

def search(query \\ __MODULE__, search_term, exclude_interests: [_|_] = excludes) do
   # ... the actual impl
end
```

Reply:
Hi Lance, thanks for this input! Lots of the changes in this PR were due to the formatter just fixing things like this one. I was trying not to introduce too many changes that would muddy the PR since the formatter did quite a few of these kinds of changes. So I'll hold off on making that change, though I like the suggestion you made.